Added new kb article load-tab-content-on-demand-net-maui (#971)
telerik/maui-docs: knowledge-base/load-tab-content-on-demand-net-maui.md

---
title: Loading Tab Content on Demand in .NET MAUI TabView
description: Learn how to load the content of TabView items on demand as the user selects tabs, improving performance for UI-heavy content in .NET MAUI applications.
type: how-to
page_title: Implementing On-Demand Content Loading in .NET MAUI TabView
slug: load-tab-content-on-demand-net-maui
tags: tabview, .net maui, on-demand loading, itemtemplate, contenttemplate
res_type: kb
ticketid: 1668168
---

## Environment

<table>
<tbody>
<tr>
<td>Product</td>
<td>TabView for .NET MAUI, <br />Version 7.0.0</td>
</tr>
</tbody>
</table>

## Description

> When using a [TabView](https://docs.telerik.com/devtools/maui/controls/tabview/overview) bound to an `ItemSource` with an `ItemTemplate`, loading UI-heavy content for each tab at page load can impact performance. To enhance application responsiveness, it's beneficial to load the content of each tab item on demand as the user selects it. This KB article also answers the following questions:

Reviewer: ```suggestion
When using a [TabView](https://docs.telerik.com/devtools/maui/controls/tabview/overview) bound to an `ItemSource` with an `ItemTemplate`, loading UI-heavy content for each tab during page load can negatively impact performance. To enhance application responsiveness, it's beneficial to load the content of each tab item on demand, only when the user selects it.

This KB article also answers the following questions:
```

- How to improve performance by loading tab content on demand in .NET MAUI TabView?
- How to delay the loading of UI-heavy content in TabView tabs until selection?
- How to use attached properties for on-demand content loading in TabView?

## Solution

> To load tab item content on demand, create a `TabViewUtils` class with two attached properties: `LoadOnDemand` and `LoadOnDemandTemplate`. 

Reviewer: I believe that each H3 section that follows is a distinct step in the process. If this is the case, list all steps in the Solution and make each list item an anchor that links to the corresponding H3, i.e.:

To load tab item content on demand, create a `TabViewUtils` class with two attached properties: `LoadOnDemand` and `LoadOnDemandTemplate`. The process consists of the following steps:

1. [Implement the Attached Properties]()
2. [Set the TabItems DataTemplate]()
3. ...
